tests/launchsecurity/secure-execution.go

```diff
@@ -0,0 +1,115 @@
+)
+
+var _ = Describe("[sig-compute]IBM Secure Execution", decorators.SecureExecution, decorators.SigCompute, decorators.RequiresS390X, func() {
+	Context("Node Labels", func() {
+		It("Should have nodes with Secure Execution Label", func() {
+			virtclient := kubevirt.Client()
+			nodes := libnode.GetAllSchedulableNodes(virtclient)
+			hasNodeWithSELabel := false
+			for _, node := range nodes.Items {
+				if _, ok := node.Labels[kubevirtv1.SecureExecutionLabel]; ok {
+					hasNodeWithSELabel = true
+					break
+				}
+			}
+			Expect(hasNodeWithSELabel).To(BeTrue())
+		})
+	})
+	Context("Ensure cluster can run Secure Execution VMs", func() {
+		var vmi *kubevirtv1.VirtualMachineInstance
+
+		const commandTimeout = 10 * time.Second
+		BeforeEach(func() {
+			By("Reading the hostkey from the 'secex-hostkey' configmap in the 'kubevirt-prow-jobs' namespace")
+			cm, err := kubevirt.Client().CoreV1().ConfigMaps("kubevirt-prow-jobs").Get(context.Background(), "secex-hostkey", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating a configmap containing the hostkey")
+			cm = libconfigmap.New(cm.Name, cm.Data)
+			cm, err = kubevirt.Client().CoreV1().ConfigMaps(testsuite.GetTestNamespace(cm)).Create(context.Background(), cm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vmi = libvmifact.NewFedora(
+				libvmi.WithConfigMapDisk(cm.Name, cm.Name),
+			)
+			// Enabling launchsecurity here won't prevent starting non-SE VMs
+			vmi.Spec.Domain.LaunchSecurity = &kubevirtv1.LaunchSecurity{}
+
+			By("Launching a non-SE VM to convert it to Secure Execution")
+			vmi = libvmops.RunVMIAndExpectLaunch(vmi, 240)
+
+			By("Logging in to non-SE VM")
+			Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+			By("Mounting the secure execution hostkey iso")
+			Expect(console.RunCommand(vmi, "mount /dev/vdb /mnt/", commandTimeout)).To(Succeed())
+
+			By("Writing the kernel cmdline to a file")
+			Expect(console.RunCommand(vmi, "cat /proc/cmdline > /tmp/parmfile.txt", commandTimeout)).To(Succeed())
+
+			By("Creating the encrypted boot image")
+			Expect(console.RunCommand(vmi, "genprotimg --no-verify -o /boot/sdboot -i /boot/vmlinuz-*.s390x -r /boot/initramfs-*.s390x.img -p /tmp/parmfile.txt -k /mnt/secex-hostkey.crt", commandTimeout)).To(Succeed())
+
+			By("Calling zipl to boot from the encrypted image")
+			Expect(console.RunCommand(vmi, "zipl -t /boot -i /boot/sdboot", commandTimeout)).To(Succeed())
+
+			By("Rebooting VM into Secure Execution mode")
+			Expect(kubevirt.Client().VirtualMachineInstance(vmi.Namespace).SoftReboot(context.Background(), vmi.Name)).To(Succeed())
+		})
+
+		It("Should launch a Secure Execution VM", func() {
+			By("Checking that the pod is scheduled on Secure Execution enabled nodes")
+			pod, err := libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pod.Spec.NodeSelector).To(HaveKeyWithValue(kubevirtv1.SecureExecutionLabel, "true"))
```

Comment:
Will the VM fail to start in case it isn't scheduled to a proper node? If so, please consider dropping this step, or covering it in unit tests.

Reply:
It will fail yes and it is also checked in a unit test. I can drop this part.